docs/user/reference/404-not-found.rst

```diff
@@ -0,0 +1,53 @@
+``404 Not Found`` pages
+=======================
+
+If you want your project to use a custom or branded ``404 Not Found`` page,
+you can put a ``404.html`` or ``404/index.html`` at the top level of your project's HTML output.
+
+How it works
+------------
+
+When our servers return a ``404 Not Found`` error,
+we check if there is a ``404.html`` or ``404/index.html`` in the root of your project's output.
+
+The following locations are checked, in order:
+
+* ``/404.html`` or ``404/index.html`` in the *current* documentation version.
+* ``/404.html`` or ``404/index.html`` in the  *default* documentation version.
+
+Tool Integration
```

Comment:
```suggestion
Tool integration
```

Reply:
This is basically the how-to part :)

I think we _should_ consider moving this to a "How to add your own custom 404 page" and referencing as a `seealso::`. I think a how-to will become really handy.

Reply:
Yea, probably. I just didn't want to make a _5th_ file :)